packages/layout/lib/src/modal/components/modal-overlay/modal-overlay.component.ts

```diff
@@ -26,12 +27,21 @@ export class ModalOverlayComponent {
 
 	constructor(
 		private ref: ElementRef
-	) {}
+	) { }
 
-	@HostListener('click', ['$event']) public clickHandler(e: MouseEvent) {
+	@HostListener('touchstart', ['$event'])
+	@HostListener('mousedown', ['$event'])
+	public mouseDownHandler(e: MouseEvent) {
+		const modal = this.ref.nativeElement.querySelector('.m-modal');
+		this.mouseDownInsideOverlay = modal && (e.target === modal || modal.contains(e.target));
+	}
+
+	@HostListener('touchend', ['$event'])
+	@HostListener('mouseup', ['$event'])
+	public mouseUpHandler(e: MouseEvent) {
 		const modal = this.ref.nativeElement.querySelector('.m-modal');
 
-		if (modal && (e.target === modal || modal.contains(e.target))) {
+		if ((modal && (e.target === modal || modal.contains(e.target))) || this.mouseDownInsideOverlay) {
 			return;
 		}
```

Comment:
het lijkt me dat je in de mouseUp this.mouseDownInsideOverlay op false moet zetten, of anders kan je na een drag in naar uit niet meer naast de modal klikken om die te sluiten.

Reply:
Hoe bedoel je? Je komt toch altijd eerst in de `mousedown` en dan in `mouseup` handler? En in de `mousedown` wordt hij telkens opnieuw gezet. Ik wil hem wel op false zetten hoor voor de zekerheid, maar lijkt mij niet echt nodig.

Reply:
ja, inderdaad, je hebt gelijk